### DISCUSSION TITLE
Use of existing Redis instance

### DISCUSSION THREAD

**Discussion by fgossel** [searxng/searxng#1855] [Q&A] [answered]
Is it possible to use an existing Redis server and just set the following free DB for url:, e.g. db=1?
As far as I understand from the source code and experiments, this is not possible. Am I right or am I doing it wrong?
After connecting I get an error "redis.exceptions.ResponseError: unknown command `ACL`, with args beginning with: `WHOAMI`".

**Comment by dalf** [comment]
> After connecting I get an error "redis.exceptions.ResponseError: unknown command ACL, with args beginning with: WHOAMI".


If this is easy for you, can you try to replace this line
https://github.com/searxng/searxng/blob/a733761288ef35ede2698af6028e82af1bf1c7bc/searx/shared/redisdb.py#L45
by
```python
        logger.info("connected redis DB")
```

Note: currently, Redis is only required for the limiter.

**Comment by fgossel** [comment]
Thanks for your quick response, this helped. Now limiter is loaded. 
```
DEBUG   searx.plugins                 : searx.plugins.limiter: loaded           
DEBUG   searx.plugins.limiter         : init limiter DB                         
INFO    searx.shared.redis            : connected redis DB 
```
I don't have much experience with Redis and wasn't sure if just pointing to another free DB was enough.

**Reply by return42** [reply]
> After connecting I get an error "redis.exceptions.ResponseError: unknown command ACL, with args beginning with: WHOAMI".

Seems you REDIS server is an older version .. available since 6.0 .. https://redis.io/commands/acl-whoami/

![grafik](https://user-images.githubusercontent.com/554536/195306707-213a67d9-301d-4ce4-b190-e0aa68298406.png)

**Reply by fgossel** [reply]
omg, you're right! I looked everywhere, but I didn't think of this. Embarrassing.
My Redis version is 5.

**Comment by fgossel** [comment, marked as answer]
OK, I can confirm that everything works as it should with Redis version 6.0. 
For lower versions of Redis, just modify as @dalf suggested.
